cli/src/swam/cli/Main.scala

```diff
@@ -118,11 +121,20 @@ object Main extends CommandIOApp(name = "swam-cli", header = "Swam from the comm
 
   val logger = consoleLogger[IO]()
 
+  def measureTime[T](t: => T): T = {
+    val start = System.nanoTime()
+    val res = t
+    val end = System.nanoTime()
+    System.err.println(s"${end - start}ns")
+    res
+  }
```

Comment:
This won't work, since your `t` here is an `IO` so the computation is suspended, and you will only measure the time it takes. Try this:
```suggestion
  def measureTime[T](io: IO[T]): IO[T] =
    for {
      start <- Clock[IO].monotonic(TimeUnit.NANOSECONDS)
      res <- io
      end <- Clock[IO]monotonic(TimeUnit.NANOSECONDS)
      _ <- logger.info(s"Execution took ${end - start}ns)
    } yield res
```

Reply:
If the computation throws an exception, time won't be logged, if you want it we can do it with other constructs, like `guarantee` but I think it is a good first version.